### PR TITLE
fix(service-worker): Re-register one signal service worker on update

### PR DESCRIPTION
**Pull request by beMaddz**
# Description

On update of app version or config, re-register one-signal service worker.

### CODE REVIEW

**Review by EnKrypt — APPROVED** [review]

**Review by ReenaSingh07 — APPROVED** [review]

**Review by Deokumarjnu — APPROVED** [review]

File: client/impl/load-service-worker.js [hunk 1]
```diff
@@ -5,31 +5,40 @@ export function registerServiceWorker({
...
 
-export function setupServiceWorkerUpdates(
-  serviceWorkerPromise,
-  app,
-  store,
-  page
-) {
+function updateOneSignalWorker(page, opts) {
+  let { config: { "theme-attributes": pageThemeAttributes = {} } = {} } = page;
+  let version = pageThemeAttributes["cache-burst"];
```
Comment (beMaddz) [inline client/impl/load-service-worker.js:22]: use const

File: client/impl/load-service-worker.js [hunk 2]
```diff
@@ -5,31 +5,40 @@ export function registerServiceWorker({
   serviceWorkerLocation = "/service-worker.js",
   navigator = global.navigator,
   mountAt = global.qtMountAt || "",
+  version = 0,
```
Comment (beMaddz) [inline client/impl/load-service-worker.js:8]: pass appVersion for custom publishers

File: client/impl/load-service-worker.js (hunk 1 above)
Comment (beMaddz) [reply client/impl/load-service-worker.js:22]: check if `version` is present else pass app.version

File: client/start.js [hunk 3]
```diff
@@ -343,7 +341,13 @@ export function startApp(renderApplication, reducers, opts) {
       const mssgSenderId = get(page, ["config", "fcmMessageSenderId"], null);
       initializeFCM(mssgSenderId);
     }
-    setupServiceWorkerUpdates(serviceWorkerPromise, app, store, page);
+
+    let { config: { "theme-attributes": pageThemeAttributes = {} } = {} } = page;
+    let version = pageThemeAttributes["cache-burst"];
```
Comment (beMaddz) [inline client/start.js:346]: check if version is present else pass app.version also use const

File: client/start.js [hunk 4]
```diff
@@ -329,9 +327,9 @@ export function startApp(renderApplication, reducers, opts) {
...
   }
 
-  return dataPromise.then((page) => doStartApp(page));
+  return dataPromise.then((page) => doStartApp(page, opts));
 
-  function doStartApp(page) {
+  function doStartApp(page, opts) {
     if (!page) {
       console &&
         console.log("Recieved a null page. Expecting the browser to redirect.");
```
Comment (shraddha-kesari) [inline client/start.js:335]: *Received

**Pull request merged**